doc/heps/hep2.md

```diff
@@ -0,0 +1,366 @@
+
+#### Supported versions
+
+Aside from certain exceptional cases, the Projects are not expected to backport changes to previous major or minor releases.
+
+#### Distribution
+
+HoloViz release managers are responsible for distributing the Projects on these platforms:
+
+- Pre-releases are distributed on [PyPI](https://pypi.org) and on the *pyviz/label/dev* channel of [Anaconda.org](https://anaconda.org).
+- Final versions are distributed on [PyPI](https://pypi.org), and on the  *conda-forge* and *pyviz* channels of [Anaconda.org](https://anaconda.org).
+
+#### Release cadence
+
+The Projects have not adopted a regular release cadence; a new release happens whenever maintainers find it is appropriate. A Project is free to adopt a regular release cadence.
+
+Major regressions in a release should be fixed and released in a new patch version as soon as possible.
+
+### Backwards compatibility
+
+The HoloViz Projects serve different purposes in the ecosystem. Some are more foundational and have been in place for a long time, for instance, Colorcet or Param. Given their position, these Projects should be treated with extra care when it comes to backwards compatibility. For instance, maintainers of these Projects should favor making breaking changes in major releases and adopting longer deprecation periods. On the other hand, some Projects are newer and have an API that is still being gradually refined (e.g. Lumen). These Projects are not expected to be as stable, they change more quickly. 
+
+Overall, the HoloViz Projects are known to be stable and their users have built this expectation. The functionalities they provide are generally not moved or removed lightly. Maintainers should aim to keep the Projects stable; moving or removing a feature from a code base **must** be motivated, in particular when it is done outside of a major release.
+
+### Deprecation policy
+
+#### Deprecation cycle guidelines
+
+The Projects can deprecate a feature in any type of final release (patch, minor, or major). However, before implementing the deprecation, they **must**:
+
+- Motivate the best they can why the feature is deprecated (e.g. old and unused API) in an issue or in the *Pull Request* (PR) deprecating the feature.
+- Reach a consensus among the maintainers of the Project on the deprecation.
+- Make sure that the PR deprecating the feature is approved by at least another maintainer, who should check that the indications below have been followed.
+
+ Maintainers have to make sure that their users are well informed of the deprecation. They **must** mention the deprecation in the following ways:
+
+- Implement a programmatic warning, if not applicable the PR deprecating the feature **must** clearly explain why:
```

Comment:
Should we not standardize on one mechanism or are we worried about the `typing_extensions` dependency?

Reply:
Given `deprecated` is a decorator I don't think is can be used in all the places you may want to use `warn`?